### PR TITLE
Add QuadDIRECT global optimizer 

### PR DESCRIPTION
**Pull request by ranjanan**

### CODE REVIEW

File: src/require.jl [hunk 1]
```diff
@@ -168,5 +168,54 @@ function __init__()
                                                 t1 - t0)
         end
     end
+    @require QuadDIRECT = "dae52e8d-d666-5120-a592-9e15c33b8d7a"
+        export QuadDIRECT
```
Comment (ChrisRackauckas) [inline src/require.jl:172]: why export?

Comment (ranjanan) [reply src/require.jl:172]: The Optim solvers are exported. So I thought I'd export this as well. We don't need to though. Would you prefer to not export this?

**Comment by ranjanan** [comment]
@ChrisRackauckas QuadDIRECT is registered in a different registry (HolyLabs). How do I make it available in the tests? Do I add a line in Travis about adding HolyLabs as a registry?

File: src/require.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/require.jl:172]: if you have to use requires, then `using QuadDIRECT` is already done on the user side, so it's all already there.

File: src/require.jl [hunk 2]
```diff
@@ -168,7 +168,7 @@ function __init__()
                                                 t1 - t0)
         end
     end
-    @require QuadDIRECT = "dae52e8d-d666-5120-a592-9e15c33b8d7a"
+    @require QuadDIRECT = "dae52e8d-d666-5120-a592-9e15c33b8d7a" begin
         export QuadDIRECT
         struct QuadDIRECT
```
Comment (ChrisRackauckas) [inline src/require.jl:173]: is that not going to cause issues being the same name as the package?

File: .travis.yml [hunk 3]
```diff
@@ -10,6 +10,9 @@ julia:
 #    - julia: nightly
 notifications:
   email: false
+script: 
+ - julia -e 'using Pkg; pkg"registry add git@github.com:HolyLab/HolyLabRegistry.git"'
```
Comment (ChrisRackauckas) [inline .travis.yml:14]: ugh that's ugly.

**Pull request merged**